pinot-spi/src/main/java/org/apache/pinot/spi/trace/RequestContext.java

```diff
@@ -217,6 +217,10 @@ default boolean isSampledRequest() {
 
   void setProcessingExceptions(List<String> processingExceptions);
 
+  Map<String, List<String>> getRequestHttpHeaders();
```

Comment:
@walterddr : what's the general guideline on adding new methods to `RequestContext`?

Reply:
good question I am not 100% sure. CC @xiangfu0 and @Jackie-Jiang